R/pkg/R/DataFrame.R

```diff
@@ -1138,6 +1138,11 @@ setMethod("collect",
                   if (!is.null(PRIMITIVE_TYPES[[colType]]) && colType != "binary") {
                     vec <- do.call(c, col)
                     stopifnot(class(vec) != "list")
+                    class(vec) <-
+                      if (colType == "timestamp")
+                        c("POSIXct", "POSIXt")
+                      else
+                        PRIMITIVE_TYPES[[colType]]
```

Comment:
by setting these instead of having it inferred - does this break any existing behavior? does any type differ because of this line of change?